File: FbxSharpTests/LayerContainerTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class LayerContainerTest : TestBase
    {
        [Test]
        public void LayerContainer_Create()
        {
            // given:

            // when:
            var lc = new FbxMesh("");

            // then:
            Assert.AreEqual(0, lc.GetSrcObjectCount());
            Assert.AreEqual(0, lc.GetDstObjectCount());
            Assert.AreEqual(0, lc.GetLayerCount());
        }

        [Test]
        public void LayerContainer_CreateLayer_CreatesNewLayer()
        {
            // given:
            var lc = new FbxMesh("");

            // require:
            Assert.AreEqual(null, lc.GetLayer(0));

            // when:
            var result = lc.CreateLayer();

            // then:
            Assert.AreEqual(0, result);
            Assert.AreNotEqual(null, lc.GetLayer(0));
        }
    }
}

File: FbxCppTests/LayerContainerTest.cpp

#include "Tests.h"

using namespace std;

void LayerContainer_Create()
{
    // given:
    FbxManager* manager = FbxManager::Create();

    // when:
    FbxLayerContainer* lc = FbxLayerContainer::Create(manager, "");

    // then:
    AssertEqual(0, lc->GetSrcObjectCount());
    AssertEqual(0, lc->GetDstObjectCount());
    AssertEqual(0, lc->GetLayerCount());
}

void LayerContainer_CreateLayer_CreatesNewLayer()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxLayerContainer* lc = FbxLayerContainer::Create(manager, "");

    // require:
    AssertEqual(NULL, lc->GetLayer(0));

    // when:
    int result = lc->CreateLayer();

    // then:
    AssertEqual(0, result);
    AssertNotEqual(NULL, lc->GetLayer(0));
}

void LayerContainerTest::RegisterTestCases()
{
    AddTestCase(LayerContainer_Create);
    AddTestCase(LayerContainer_CreateLayer_CreatesNewLayer);
}

